tests/unittest/_torch/modules/moe/test_moe_backend.py

```
@@ -480,6 +487,183 @@ def test_marlin_override_quant_config_degrades_per_layer():
+        pytest.param(QuantAlgo.W4A16_MXFP4, id="w4a16_mxfp4"),
+    ],
+)
+def test_trtllm_gen_situ_rejects_quant_algos_without_fused_cubins(quant_algo) -> None:
+    """There is no standalone SiTu activation kernel.
+
+    SiTu exists only as a fused FC1 epilogue, in the NVFP4 (group-16) and
+    W4A8_MXFP4_MXFP8 (group-32) cubin families. Anything else has to be
+    rejected at construction rather than silently resolving to SwiGLU, which
+    is structurally wrong output that no shape check would catch.
+    """
+    with pytest.raises(ValueError, match="requires one of .* quantization"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)